runtime/rasdump/dmpsup.c

```diff
@@ -577,7 +578,32 @@ configureDumpAgents(J9JavaVM *vm)
 		}
 		if ( IS_CONSUMABLE(j9vm_args, xdumpIndex) && !IS_CONSUMED(j9vm_args, xdumpIndex) )
 		{
-			GET_OPTION_VALUE(xdumpIndex, ':', &optionString);
+			/* Handle mapped tool dump options */
+			if (HAS_MAPPING(j9vm_args, xdumpIndex)) {
+				char *toolCursor = strstr(MAPPING_J9NAME(j9vm_args, xdumpIndex), ":tool") + 1;
+				if (NULL != toolCursor) {
+					/* The mapped option specifies the tool command to run after the equals */
+					char *optionValue = NULL;
+					GET_OPTION_VALUE(xdumpIndex, '=', &optionValue);
+					
+					/* Construct optionString by combining the J9 tool dump command with the mapped option */
+					int toolCursorLength = strlen(toolCursor);
+					int optionValueLength = strlen(optionValue);
+					optionString = (char *) j9mem_allocate_memory(toolCursorLength + optionValueLength + 1, OMRMEM_CATEGORY_VM);
+
+					if (NULL == optionString) {
+						j9tty_err_printf(PORTLIB, "Storage for mapped tool dump optionString not available, unable to process option\n");
+						return J9VMDLLMAIN_FAILED;
+					} else {
+						strcpy(optionString, toolCursor);
+						strcat(optionString + toolCursorLength, optionValue);
+						/* Record original optionString pointer for freeing mapped tool dump optionString */
+						optionStringOriginal = optionString;
```

Comment:
This code is in a loop. If more than one option is mapped, there is a memory leak.